packages/flow/lib/AstroCase.d.ts

```diff
@@ -1,21 +1,17 @@
-export type Props<T> = 'of' extends keyof T
-	? T['of'] extends (string | number | bigint | boolean | symbol | null | undefined)
-		? {
-			of: T
-		}
-	: T['of'] extends ((value: unknown) => boolean)
-		? {
-			of: T
-		}
-	: T['of'] extends object
-		? {
-			of: T
-		}
-	: never
-: 'default' extends keyof T
-	? {
-		default: "" | true
-	}
-: any
 
-export default function Case<T>(props: Props<T>): any
+type AllowedTypes = string | null | bigint | Record<PropertyKey, unknown> | undefined | number | boolean
+
+type FilledCase = {
+    of: AllowedTypes |  ((value:unknown)=> boolean);
+    default?:never
+}
+
+type DefaultCase = {
+    default:true;
+    of?:never
+    }
+type Case =  FilledCase |  DefaultCase
```

Comment:
This typing ensures that the user can only choose between the `of=` or the `default=` attribute when it comes to typing a component